### PR TITLE
Test the shared version of the UMF library on Windows

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Test the shared version of the UMF library on Windows.

### Checklist
- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

**Comment by PatKamin** [comment]
Please, try running the tests with the changes from #401

**Comment by lukaszstolarczuk** [comment]
you can rebase, Patryk's change has been merged now :)

**Comment by ldorau** [comment]
> you can rebase, Patryk's change has been merged now :)

@lukaszstolarczuk Done

File: .github/workflows/basic.yml [hunk 1]
```diff
@@ -181,7 +181,7 @@ jobs:
         build_type: [Debug, Release]
         compiler: [{c: cl, cxx: cl}]
         pool_tracking: ['ON', 'OFF']
-        shared_library: ['OFF']
+        shared_library: ['ON', 'OFF']
         level_zero_provider: ['ON']
         include:
```
Comment (lukaszstolarczuk) [inline .github/workflows/basic.yml:186]: you have to add shared_library into all `include`s

File: .github/workflows/basic.yml [hunk 2]
```diff
@@ -181,7 +181,7 @@ jobs:
         build_type: [Debug, Release]
         compiler: [{c: cl, cxx: cl}]
         pool_tracking: ['ON', 'OFF']
-        shared_library: ['OFF']
+        shared_library: ['ON', 'OFF']
```
Comment (lukaszstolarczuk) [inline .github/workflows/basic.yml:184]: perhaps you should also enable windows builds in `proxy_lib.yml`...?

File: .github/workflows/basic.yml (hunk 1 above)
Comment (ldorau) [reply .github/workflows/basic.yml:186]: Added to all Windows builds. The rest is out of scope of this PR.

File: .github/workflows/basic.yml (hunk 2 above)
Comment (ldorau) [reply .github/workflows/basic.yml:184]: It is impossible - there is no `LD_PRELOAD` mechanism on Windows ...

File: .github/workflows/basic.yml (hunk 1 above)
Comment (lukaszstolarczuk) [reply .github/workflows/basic.yml:186]: heh, this one is actually all we needed

**Review by lukaszstolarczuk — COMMENTED** [review]
LGTM 👍

**Review by bratpiorka — APPROVED** [review]

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**